src/test/text_streamer_test.cpp

```diff
@@ -0,0 +1,132 @@
+    auto tokens = tokenizer->encode("TEST");
+    for (size_t i = 0; i < tokens.get_size(); i++) {
+        std::optional<std::string> rval = this->streamer->put(static_cast<int64_t*>(tokens.data())[i]);
+        EXPECT_FALSE(rval.has_value());
+    }
+}
+
+TEST_F(TextStreamerTest, valueReturned) {
+    std::string testPrompt = "TEST\n";
+    auto tokens = tokenizer->encode(testPrompt);
+    for (size_t i = 0; i < tokens.get_size(); i++) {
+        std::optional<std::string> rval = this->streamer->put(static_cast<int64_t*>(tokens.data())[i]);
+        if (i < tokens.get_size() - 1) {
+            EXPECT_FALSE(rval.has_value());
+        } else {
+            EXPECT_TRUE(rval.has_value());
+            EXPECT_EQ(rval.value().compare(testPrompt), 0);
+        }
+    }
+}
+
+TEST_F(TextStreamerTest, valueReturnedCacheNotCleared) {
+    std::string testPrompt1 = "TEST";
+    auto tokens = tokenizer->encode(testPrompt1);
+    for (size_t i = 0; i < tokens.get_size(); i++) {
+        std::optional<std::string> rval = this->streamer->put(static_cast<int64_t*>(tokens.data())[i]);
+        EXPECT_FALSE(rval.has_value());
+    }
+    std::string testPrompt2 = "TEST\n";
+    tokens = tokenizer->encode(testPrompt2);
+    for (size_t i = 0; i < tokens.get_size(); i++) {
+        std::optional<std::string> rval = this->streamer->put(static_cast<int64_t*>(tokens.data())[i]);
+        if (i < tokens.get_size() - 1) {
+            EXPECT_FALSE(rval.has_value());
+        } else {
+            EXPECT_TRUE(rval.has_value());
+            EXPECT_EQ(rval.value().compare(testPrompt1 + testPrompt2), 0);
+        }
+    }
+}
+
+TEST_F(TextStreamerTest, valueReturnedCacheCleared) {
+    std::string testPrompt = "TEST\n";
+    auto tokens = tokenizer->encode(testPrompt);
+    for (size_t i = 0; i < tokens.get_size(); i++) {
+        std::optional<std::string> rval = this->streamer->put(static_cast<int64_t*>(tokens.data())[i]);
+        if (i < tokens.get_size() - 1) {
+            EXPECT_FALSE(rval.has_value());
+        } else {
+            EXPECT_TRUE(rval.has_value());
+            EXPECT_EQ(rval.value().compare(testPrompt), 0);
+        }
+    }
+    tokens = tokenizer->encode(testPrompt);
+    for (size_t i = 0; i < tokens.get_size(); i++) {
+        std::optional<std::string> rval = this->streamer->put(static_cast<int64_t*>(tokens.data())[i]);
+        if (i < tokens.get_size() - 1) {
+            EXPECT_FALSE(rval.has_value());
+        } else {
+            EXPECT_TRUE(rval.has_value());
+            EXPECT_EQ(rval.value().compare(testPrompt), 0);
+        }
+    }
+}
+
+TEST_F(TextStreamerTest, valueReturnedTextWithSpaces) {
+    std::string testPrompt = "TEST TEST TEST TEST";
+    auto tokens = tokenizer->encode(testPrompt);
+    for (size_t i = 0; i < tokens.get_size(); i++) {
+        std::optional<std::string> rval = this->streamer->put(static_cast<int64_t*>(tokens.data())[i]);
+        size_t numberOfTokensBeforeValueReturned = 2;
```

Comment:
The same comment as above - it is not clear to me which tokens we talk about, what are the steps in this test. Once we have it I will be able to understand what we test here